packages/inventory/src/containers/ListView/utils.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import { Spin, Alert } from 'antd';
+import { ColumnsType, ColumnType } from 'antd/lib/table/interface';
+import { INVENTORY_SERVICE_POINT_PROFILE_VIEW, TableColumnsNamespace } from '../../constants';
+import { Link } from 'react-router-dom';
+import {
+  LOADING_ELLIPSIS,
+  ACTIONS_TH,
+  SERVICE_POINT_TH,
+  TYPE_TH,
+  LOCATION_TH,
+  SERVICE_POINT_ID_TH,
+  FETCHING_LOCATIONS,
+  FETCHING_LOCATIONS_DESCRIPTION,
+} from '../../lang';
+import { TreeNode } from '@opensrp/location-management';
+
+/** Describes how the data will passed to the table */
+export interface TableData {
+  key: string;
+  type: string;
+  serviceName: string;
+  location: string;
+  servicePointId: string;
+}
+
+/** component rendered in the action column of the table */
+
+export const ActionsColumnCustomRender: ColumnType<TableData>['render'] = (record) => {
+  return (
+    <>
+      <Link to={`${INVENTORY_SERVICE_POINT_PROFILE_VIEW}/${record.uniqueId}`}>View</Link>
+    </>
+  );
+};
+
+/** product Catalogue table columns */
+export const columns: ColumnsType<TableData> = [
+  {
+    title: SERVICE_POINT_TH,
+    dataIndex: 'serviceName',
+    key: `${TableColumnsNamespace}-${SERVICE_POINT_TH}`,
+    defaultSortOrder: 'descend',
+    sorter: (rec1, rec2) => {
+      if (rec1.serviceName > rec2.serviceName) {
+        return -1;
+      } else if (rec1.serviceName < rec2.serviceName) {
+        return 1;
+      } else {
+        return 0;
+      }
+    },
+  },
+  {
+    title: TYPE_TH,
+    dataIndex: 'type',
+    key: `${TableColumnsNamespace}-${TYPE_TH}`,
+  },
+  {
+    title: LOCATION_TH,
+    dataIndex: 'location',
+    key: `${TableColumnsNamespace}-${LOCATION_TH}`,
+  },
+  {
+    title: SERVICE_POINT_ID_TH,
+    dataIndex: 'servicePointId',
+    key: `${TableColumnsNamespace}-${SERVICE_POINT_ID_TH}`,
+  },
+  {
+    title: ACTIONS_TH,
+    key: `${TableColumnsNamespace}-actions`,
+    render: ActionsColumnCustomRender,
+    width: '20%',
+  },
+];
+
+/** util component shown when there is a pending promise */
+
+export const ServicePointsLoading = ({
+  message = FETCHING_LOCATIONS,
+  description = FETCHING_LOCATIONS_DESCRIPTION,
+}) => {
+  return (
+    <Spin tip={LOADING_ELLIPSIS}>
```

Comment:
Is this a new UI feature? I haven't seen it before in our code

Reply:
its the spinner, has a bit info on what is loading.

Reply:
@p-netm Ok..its great but no other component uses such a spinner, don't you think it will introduce inconsistencies in the UX?

Reply:
I think only the styles would probably need to be reconciled with what the rest of opensrp is using, that is centering it and maybe adding the size. This will be used across all of inventory pages.

Reply:
https://github.com/OpenSRP/web/issues/346

Reply:
@p-netm Ok..but this particular spinner is also showing an alert right? The spinners we have been using have no alerts or tips...we have been using a simple spinner

Reply:
I will add that as a discussion item to above issue, since we have such a spinner in several modules already.

Reply:
@p-netm is it the same spinner like the one used in the other pages? The other pages do not have info when loading why do we need it here? What kind of info is it?

Reply:
@dubdabasoduba https://github.com/OpenSRP/web/issues/346#issuecomment-762652844, just some text on what is being loaded.